### ISSUE TITLE
feature: design and develop a UI to generate tailcall configurations

### ISSUE THREAD

**Issue by tusharmath** [tailcallhq/tailcallhq.github.io#373]
Currently, Tailcall Configurations can be written in JSON, YML, and GraphQL formats. Internally they are parsed in [config.rs](https://github.com/tailcallhq/tailcall/blob/main/src/core/config/config.rs)

Config is pretty core to Tailcall and we would like to enhance the experience by using a web interface. Create a form that allows developers to quickly create, update and download Tailcall Configurations.

**Technical Requirements**
- Make sure the generated configurations are always correct.
- Use searchable dropdowns wherever possible instead of text boxes.
- Ensure the UI automatically updates with changes in the [schema](https://github.com/tailcallhq/tailcall/blob/main/generated/.tailcallrc.schema.json)
- Ensure the schema is loaded dynamically from the above URL.
- The page should be integrated into https://tailcall.run/playground

**Design Requirements**
- UI Should follow the current design theme and look good.
- It should be mobile friendly.

**Update 1**
- Deploy this UI on [https://tailcall.run/app/config](https://tailcall.run/app/config)

**Comment by harshtech123** [comment]
/attempt #373


<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clyv5yiks0002mr09xl16587d/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by mehulmathur16** [comment]
/attempt #373

<div id="algora-attempt" />

| [Algora profile](https://console.algora.io/@/mehulmathur16) | Completed bounties | Tech | Active attempts | Options |
| --- | --- | --- | --- | --- |
| @mehulmathur16 | &nbsp;&nbsp;&nbsp;16 tailcallhq bounties <br/> + 8 bounties from 4 projects | <div align="center">TypeScript, Rust, <br />HTML & more</div> | <div align="center"></div> | [Cancel attempt](https://console.algora.io/api/bounties/clyv5yiks0002mr09xl16587d/cancel-attempt) |

**Comment by Bhavyajain21** [comment]
/attempt #373

<div id="algora-attempt" />

| [Algora profile](https://console.algora.io/@/Bhavyajain21) | Completed bounties | Tech | Active attempts | Options |
| --- | --- | --- | --- | --- |
| @Bhavyajain21 | &nbsp;&nbsp;&nbsp;3 tailcallhq bounties <br/> + 11 bounties from 2 projects | <div align="center">TypeScript, Rust, <br />JavaScript & more</div> | <div align="center">﹟360</div> | [Cancel attempt](https://console.algora.io/api/bounties/clyv5yiks0002mr09xl16587d/cancel-attempt) |

**Comment by tusharmath** [comment]
I would recommend people collaborate on this issue together and work towards a common goal.

**Comment by Bhavyajain21** [comment]
@mehulmathur16 Let's collab on this?

**Comment by mehulmathur16** [comment]
Sure @Bhavyajain21 👍

**Comment by harshtech123** [comment]
on it
{UI Should follow the current design theme and look good.
It should be mobile friendly}.

**Comment by webbdays** [comment]
@tusharmath 
how is this? ok?

![Screen Shot 2024-07-31 at 22 07 46](https://github.com/user-attachments/assets/293249b6-65d5-440c-adfc-6379acab9193)

![Screen Shot 2024-07-31 at 22 07 41](https://github.com/user-attachments/assets/0bf912e3-8764-437d-a20d-22e8330ff65f)

**Comment by webbdays** [comment]
![laptop](https://github.com/user-attachments/assets/afade5c3-fda6-4ef6-ada7-6514c45883f3)

**Comment by webbdays** [comment]
Healthy competition. 😄😆

**Comment by webbdays** [comment]
Hi,
Code is ready.
if ui is ok.
I will push and make a pr.

**Comment by tusharmath** [comment]
# ATTENTION!!!

Thanks everyone! I love what you guys have created from an engineering point of view however I think we need some professional help in terms of design 🙈. Im going to get it from outside and try to share the figma URLs here. Until then we will just keep the issue on hold. Thanks for understanding 🙏

**Comment by 35C4n0r** [comment]
Throwing my hat in too once the designs are done!

**Comment by yasamoka** [comment]
Is the bounty open? Can one make an attempt? Thanks.

**Comment by icorbrey** [comment]
@tusharmath I'd like to try my hand at getting a design together, would I be eligible to receive part of the bounty if I contribute to this?

**Comment by tusharmath** [comment]
@icorbrey can u connect with me on discord?

**Comment by icorbrey** [comment]
@tusharmath Yep! I @'d you in #contributors

**Comment by icorbrey** [comment]
/attempt #373

<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clyv5yiks0002mr09xl16587d/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by icorbrey** [comment]
@tusharmath still working out the kinks in the interactive bits, but this is what I've got cooking up for desktop UI:

![Image](https://github.com/user-attachments/assets/a0bad67d-8b3d-4acb-89ed-4e019ac2b051)

And for mobile:

![Image](https://github.com/user-attachments/assets/854c0a7f-3529-4d99-905c-813363b1f20f)

Shallow objects will have expandable bits and deep objects will show in a slideover. Any feedback so far to take into account? Obviously I'll be taking some time to polish up the appearance after the interactions are down

**Comment by edilson258** [comment]
@tusharmath still open??

**Comment by icorbrey** [comment]
Probably not @edilson258, Tushar seems to be completely focused on Forge at the moment

**Comment by yasamoka** [comment]
I think this bounty should be closed out of respect to those who may attempt and get no response in return.

**Comment by aadarsh-nagrath** [comment]
Yeah please close this bounty

**Comment by mike123456789009** [comment]
Hi, I did a bounded prototype spike for this bounty before opening a PR because I saw the earlier hold note and wanted to avoid adding noise if the direction has changed.

Prototype status:
- added a schema-driven Config Builder page under `/app/config`
- loads the Tailcall config JSON schema and renders grouped fields
- generates typed JSON output as the user edits values
- added a Playground entry point linking to the Config Builder
- local production build passes and desktop/mobile visual QA passed

Before I open the PR and claim the bounty, can a maintainer confirm whether #373 is still wanted in this shape? If the current product direction is different, I can adapt the prototype to the preferred route rather than submitting stale work.

**Comment by poofeth** [comment]
/attempt #373

I opened PR #598 with a schema-driven `/app/config` generator implementation. It loads the generated Tailcall schema, renders searchable controls for key sections, supports repeatable `@link` entries, and exports JSON/YAML/GraphQL configs. Local validation on the PR includes `npm run build` and Prettier checks for the changed config-generator files.

**Comment by belcrod5** [comment]
Hi maintainers, quick check on #373 status.

As of May 12, 2026, I see open PRs #595 and #599, and an earlier hold note in this issue. Is this bounty still active and accepting new contributions? If yes, I can focus only on uncovered gaps to avoid duplicate work.

**Comment by jamilahmadzai** [comment]
/attempt #373

Plan: build a schema-driven `/app/config` page that loads the Tailcall runtime schema at runtime, renders the available config sections from that schema, validates/prunes the generated config before export, supports JSON/YAML/GraphQL output plus import/copy/download, and adds a small entry point from `/playground` without adding new runtime dependencies.

**Comment by chinook1001** [comment]
Submitted PR: https://github.com/tailcallhq/tailcallhq.github.io/pull/614

/claim #373

Validation summary:
- `npm run build` passed locally.
- Targeted Prettier format check passed for the changed config generator/playground files.
- Desktop and narrow viewport browser checks passed locally for `/app/config`.

**Comment by 18166714330cl-maker** [comment]
/attempt #373

Plan: inspect the existing playground/config code and the Tailcall schema, then split the work into the smallest useful slice: dynamic schema loading, form rendering, and export/download. I will only open a PR if I can keep the first change scoped and verifiable.

**Comment by goodnight-eriri** [comment]
/attempt #373

Hi maintainers, I’d like to work on this if it’s still available. Since this issue is older, could you confirm that the bounty is still active and that `/app/config` is still the desired target?

My proposed scope: build a schema-driven config generator UI, load the schema dynamically, use searchable controls where appropriate, support exporting/downloading valid JSON/YAML/GraphQL configs, match the current Tailcall design, and make it mobile friendly.

**Comment by patchharbor-labs** [comment]
Hi! Before starting work, could you confirm whether the $500 bounty is still active? I see open PR #626 covering much of the scope. If additional work is still wanted, which acceptance gaps should a new contribution address? I’d like to avoid duplicating an active effort.
